### ISSUE TITLE
Failed to run k8s-device-plugin with containerd

### ISSUE THREAD

**Issue by inductor** [NVIDIA/nvidia-container-runtime#127]
Hello, 

I am trying to use my GeForce GTX 1660 on Kubernetes and `k8s-device-plugin` fails due to the following error.

```
  Warning  Failed     18s (x3 over 36s)  kubelet            Error: failed to create containerd task: OCI runtime create failed: container_linux.go:370: starting container process caused: process_linux.go:459: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: nvidia-container-cli: mount error: open failed: /sys/fs/cgroup/devices/system.slice/containerd.service/kubepods-besteffort-pod098b3bb9_8f7c_463a_adec_4a2ce0901848.slice/devices.allow: no such file or directory: unknown
```

Here's my containerd setting

```toml
    [plugins."io.containerd.grpc.v1.cri".containerd]
      snapshotter = "overlayfs"
      default_runtime_name = "nvidia"
      no_pivot = false
      disable_snapshot_annotations = true
      discard_unpacked_layers = false
      [plugins."io.containerd.grpc.v1.cri".containerd.default_runtime]
        runtime_type = ""
        runtime_engine = ""
        runtime_root = ""
        privileged_without_host_devices = false
        base_runtime_spec = ""
      [plugins."io.containerd.grpc.v1.cri".containerd.untrusted_workload_runtime]
        runtime_type = ""
        runtime_engine = ""
        runtime_root = ""
        privileged_without_host_devices = false
        base_runtime_spec = ""
      [plugins."io.containerd.grpc.v1.cri".containerd.runtimes]
        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
          runtime_type = "io.containerd.runc.v2"
          runtime_engine = ""
          runtime_root = ""
          privileged_without_host_devices = false
          base_runtime_spec = ""
          [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
            SystemdCgroup = true
        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.nvidia]
          runtime_type = "io.containerd.runc.v2"
          runtime_engine = ""
          runtime_root = ""
          privileged_without_host_devices = false
          base_runtime_spec = ""
          [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.nvidia.options]
            BinaryName = "nvidia-container-runtime"
```

Here's the result of `nvidia-smi`

```
# nvidia-smi
Sat Dec 12 15:56:40 2020
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 455.45.01    Driver Version: 455.45.01    CUDA Version: 11.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  GeForce GTX 1660    On   | 00000000:01:00.0 Off |                  N/A |
| 35%   30C    P8    14W / 120W |     13MiB /  5944MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A       967      G   /usr/lib/xorg/Xorg                  8MiB |
|    0   N/A  N/A      1044      G   /usr/bin/gnome-shell                2MiB |
+-----------------------------------------------------------------------------+
```

Kubernetes version: 1.20.0 with Kubeadm
Containerd version: 1.4.3

If there's anything unclear please let me know.

Thanks!

**Comment by klueska** [comment]
This appears to be triggered when using the `systemd` cgroup driver with `containerd`.  Also reported here: https://github.com/NVIDIA/libnvidia-container/issues/119

We plan to have a fix for this in the 1.4.0 release of `libnvidia-container` coming out in January.

**Comment by inductor** [comment]
@klueska Hello! I was just wondering if it's coming anytime soon :)

**Comment by klueska** [comment]
The fix for this has now been merged and will be included in a patch release of libnvidia-container coming next week.
https://gitlab.com/nvidia/container-toolkit/libnvidia-container/-/merge_requests/49

**Comment by inductor** [comment]
@klueska Hello. It doesn't seem to be like there's no release yet. https://gitlab.com/nvidia/container-toolkit/libnvidia-container/-/tags

**Comment by elezar** [comment]
Hi @inductor. The change Kevin mentioned was shipped as part of the 1.3.2 release: https://github.com/NVIDIA/libnvidia-container/releases/tag/v1.3.2
